### DISCUSSION TITLE
IDateTime interface usage

### DISCUSSION THREAD

**Discussion by i02coroj** [jasontaylordev/CleanArchitecture#748] [General]
Hi all.
I'd like to understand what's the real purpose of IDateTime interface? The implementation for it just gives back a DateTime.Now value.
I understand this is an extra level of abstraction but wondering if it's worth in this case. What other implementations of it could we have?

Or perhaps I'm missing the point?

Thans in advance for your thoughts!

**Comment by danielmackay** [comment]
`IDateTime` provides you with a couple of advantages. Using the interface throughout your app allows you to consistently use local or UTC time (depending on your need).

The other advantage is that it allows you to mock the service for testing. This can be very valuable if you need to test logic that depends on a specific date or time.

**Reply by i02coroj** [reply]
I see. That makes perfect sense.
Thanks.
